### PR TITLE
Clean Code / Meaningful Names / Don't Add Gratuitous Context & Add Meaningful Context

### LINKED ISSUE #27: Book Club: Document our use of "Clean Code: Add meaningful context / Don’t add gratuitous context"
Add documentation to clarify the content of the related section of the Clone Code book in this repository as it relates to AndcultureCode.  Add code examples when and where it makes the most sense in the languages we use.  All content should make use of or follow the patterns established in [`CODE_CONVENTIONS.MD`](https://github.com/AndcultureCode/AndcultureCode/blob/main/CODE-CONVENTIONS.md)

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Copy over skeleton of Clean Code markdown file and add section for gratuitous & meaningful context

Closes #27 Book Club: Document our use of "Clean Code: Add meaningful context / Don’t add gratuitous context"

- [x] Related GitHub issue(s) linked in PR description
- [-] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [-] Code builds cleanly (no _additional_ warnings or errors)
- [-] Manually tested
- [-] Automated tests are passing
- [-] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Review by KevinBusch — APPROVED** [review]
Looks good!

**Comment by KevinBusch** [comment]
After rereading the section, I'm wondering if this should also incorporate some context around naming of classes and not just naming of instances?

**Comment by brandongregoryscott** [comment]
> After rereading the section, I'm wondering if this should also incorporate some context around naming of classes and not just naming of instances?

Hmm. I'll have to think on this to provide an example. Naming of entities/classes has more grey area IMO, so contriving an example may be trickier

File: CLEAN-CODE.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Clean Code
+
+These are Andculture's code standards based on Robert Martin's _Clean Code_.
+
+## Meaningful Names
+
+### Don't Add Gratuitous Context
+
+When a variable name can be inferred from the context of the class, interface or other actor you're in, prefixing the variable with the additional context adds no value. Extra context should only be added for actors that differ from the domain/context.
```
Comment (wintondeshong) [inline CLEAN-CODE.md:9]: 👍

**Review by wintondeshong — APPROVED** [review]
@brandongregoryscott thanks for writing this all out. A very important point that will improve the signal to noise ratio in our code base. Appreciate the examples

File: CLEAN-CODE.md [hunk 2]
```diff
@@ -1,41 +1,158 @@
...
+    private readonly IRepositoryReadConductor<Chapter> _chapterReadConductor;
+    private readonly ILogger<ChaptersController> _logger;
+    private readonly IMapper _mapper;
+
+    #endregion Private Properties
+
+    #region Constructor
+
+    public ChaptersController(
+        IRepositoryReadConductor<Chapter> chapterReadConductor,
```
Comment (jstevenson) [inline CLEAN-CODE.md:66]: Like on line 56, do you want to call out each redundant name prefix in a comment? It makes looking at the code quickly easier to see issues.

**Review by jstevenson — APPROVED** [review]

File: CLEAN-CODE.md (hunk 2 above)
Comment (brandongregoryscott) [reply CLEAN-CODE.md:66]: Good call 👍 I agree, pointing it out in each instance would help bring attention to it.

**Comment by brandongregoryscott** [comment]
@KevinBusch I'm going to merge this for now - if we can think of additional examples to add, let's follow up with another PR.

**Pull request merged**